dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -57,72 +58,124 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
 
     void AppNotificationManager::Register()
     {
-        winrt::guid storedComActivatorGuid{ GUID_NULL };
-        if (!PushNotificationHelpers::IsPackagedAppScenario())
+        HRESULT hr{ S_OK };
+        std::wstring appId{};
+
+        auto logTelemetry{ wil::scope_exit([&]() {
+            AppNotificationTelemetry::LogRegister(hr, appId);
+        }) };
+
+        try
         {
-            std::wstring notificationId{ RetrieveNotificationAppId() };
-            if (!AppModel::Identity::IsPackagedProcess())
+            winrt::guid storedComActivatorGuid{ GUID_NULL };
+            if (!PushNotificationHelpers::IsPackagedAppScenario())
             {
-                THROW_IF_FAILED(PushNotifications_RegisterFullTrustApplication(notificationId.c_str(), GUID_NULL));
+                appId = RetrieveNotificationAppId();
```

Comment:
This call needs to be moved before the IsPacakgedAppScenario check. Otherwise we won't populate appId when we hit the else case right?

Reply:
appId is now initialized at construction